stockflux-securities-master/src/reducers/securitiesTableReducer.js

```diff
@@ -0,0 +1,32 @@
+import { TableState } from "../enums";
+
+export const initialTableState = {
+  fetchStatus: TableState.LOADING,
+  hasErrors: false,
+  messages: []
+};
+
+export function securitiesTableReducer(state, action) {
+  switch (action.type) {
+    case TableState.LOADING:
+      return { ...state, fetchStatus: TableState.LOADING, messages: [] };
+    case TableState.DELETING:
+      return { ...state, fetchStatus: TableState.DELETING, messages: [] };
+    case TableState.UPDATING:
+      return { ...state, fetchStatus: TableState.UPDATING, messages: [] };
+    case TableState.ERROR:
+      return {
+        ...state,
+        fetchStatus: null,
```

Comment:
I don't think we should set `fetchStatus` to null, especially if we are using enum elsewhere. Maybe enum should be changed to incorporate a case for this (not not null).

Reply:
My reasoning was that when the error or success has occured, the table has stopped loading or deleting. If after loading the fetch status doesn't get set to null it remains loading and the loading spinner remains shown which is not what you want

Reply:
Ah I see, so instead of null you think I should call it `"completed"` or `"Inactive"` ?

Reply:
I've added this enum now